### PR TITLE
Remove V1 obsolete code from the monitoring appliance

### PR DESCRIPTION
**Pull request by melutzky**
Remove V1 obsolete code from the monitoring appliance

### CODE REVIEW

File: test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs [hunk 1]
```diff
@@ -1,246 +0,0 @@
...
-    public class AlertPresentationTests
-    {
-        private const string SmartDetectorName = "smartDetectorName";
-
-        [TestInitialize]
-        public void TestInitialize()
-        {
-        }
-
-#pragma warning disable CS0612 // Type or member is obsolete; Task to remove obsolete code #1312924
```
Comment (EranGon) [inline test/analysis/SmartDetectorsAnalysisTests/AlertPresentationTests.cs]: >warning disable CS0612 [](start = 8, length = 22)

find this in all of the code - make sure nothing is left

File: src/SDK/SmartDetectorsSDK/AlertPresentation/AlertPresentationPropertyAttribute.cs [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-            }
-
-            this.Section = section;
-            this.Title = title;
-        }
-
-        /// <summary>
-        /// Gets the section in which the property will be presented.
-        /// </summary>
-        public AlertPresentationSection Section { get; }
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/AlertPresentation/AlertPresentationPropertyAttribute.cs]: >AlertPresentationSection [](start = 15, length = 24)

remove this

File: src/SDK/SmartDetectors.Extensions/AlertExtensions.cs [hunk 3]
```diff
@@ -140,65 +133,6 @@ public static ContractsAlert CreateContractsAlert(this Alert alert, SmartDetecto
...
-
-        /// <summary>
-        /// Creates an <see cref="AlertPropertyLegacy"/> based on an alert presentation V1 property
-        /// </summary>
-        /// <param name="alert">The alert</param>
-        /// <param name="presentationAttribute">The attribute defining the presentation V1 of the alert property</param>
-        /// <param name="queryRunInfo">The query run information</param>
-        /// <param name="propertyStringValue">The property string value</param>
-        /// <returns>An <see cref="AlertPropertyLegacy"/></returns>
-        private static AlertPropertyLegacy CreateAlertPropertyLegacy(Alert alert, AlertPresentationPropertyAttribute presentationAttribute, QueryRunInfo queryRunInfo, string propertyStringValue)
```
Comment (EranGon) [inline src/SDK/SmartDetectors.Extensions/AlertExtensions.cs]: >QueryRunInfo [](start = 140, length = 12)

remove this class as well

Comment (melutzky) [reply src/SDK/SmartDetectors.Extensions/AlertExtensions.cs]: As we discussed, I'll remove this class after publishing the new NuGet

---
In reply to: [242828623](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r242828623) [](ancestors = 242828623)

Comment (aviled) [reply src/SDK/SmartDetectors.Extensions/AlertExtensions.cs]: why?

---
In reply to: [244528099](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r244528099) [](ancestors = 244528099,242828623)

Comment (melutzky) [reply src/SDK/SmartDetectors.Extensions/AlertExtensions.cs]: @ said it's not in use anymore

---
In reply to: [244558940](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r244558940) [](ancestors = 244558940,244528099,242828623)

Comment (melutzky) [reply src/SDK/SmartDetectors.Extensions/AlertExtensions.cs]: This class (QueryRunInfo) is part of a different repository. I'll publish a new NuGet of the monitoring appliance after this PR and then I'll update the NuGet in the other repository and remove the QueryRunInfo class from that repository.

---
In reply to: [244566473](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r244566473) [](ancestors = 244566473,244565145,244558940,244528099,242828623)

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 4]
```diff
@@ -75,15 +74,20 @@ public AnalysisServicesFactory(IExtendedTracer tracer, IHttpClientWrapper httpCl
         /// <returns>The telemetry data client, that can be used to run queries on log analytics workspaces.</returns>
         public async Task<ITelemetryDataClient> CreateLogAnalyticsTelemetryDataClientAsync(IReadOnlyList<ResourceIdentifier> resources, CancellationToken cancellationToken)
         {
+            // Verify that there are no Application Insights resources
+            if (resources.Any(resource => resource.ResourceType == ResourceType.ApplicationInsights))
+            {
+                throw new TelemetryDataClientCreationException($"Telemetry client creation failed - resources shouldn't contain resources of the type {ResourceType.ApplicationInsights}");
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: >TelemetryDataClientCreationException [](start = 26, length = 36)

naming - this is very general, it's preferable to name exceptions so the actual error is clear from the class name

my suggestion: InvalidResourcesTypesException

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 5]
```diff
@@ -75,15 +74,20 @@ public AnalysisServicesFactory(IExtendedTracer tracer, IHttpClientWrapper httpCl
...
+            }
+
             // Mark that a log signal was used to create the alert
             this.UsedLogAnalysisClient = true;
 
             // Get the query run info, and verify it
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: >query run info [](start = 23, length = 14)

OMG!

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 6]
```diff
@@ -93,17 +97,34 @@ public async Task<ITelemetryDataClient> CreateLogAnalyticsTelemetryDataClientAsy
...
         /// <returns>The telemetry data client, that can be used to run queries on Application Insights.</returns>
-        public async Task<ITelemetryDataClient> CreateApplicationInsightsTelemetryDataClientAsync(IReadOnlyList<ResourceIdentifier> resources, CancellationToken cancellationToken)
+        public Task<ITelemetryDataClient> CreateApplicationInsightsTelemetryDataClientAsync(IReadOnlyList<ResourceIdentifier> resources, CancellationToken cancellationToken)
         {
+            // Verify that there are resources
+            if (!resources.Any())
+            {
+                throw new TelemetryDataClientCreationException("No resources provided");
+            }
+
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: you don't need this - the next check covers this as well

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs (hunk 4 above)
Comment (EranGon) [reply src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: or InvalidResourcesException so you can reuse the exception for the case where the resource list is empty

---
In reply to: [244680169](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r244680169) [](ancestors = 244680169)

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 7]
```diff
@@ -143,23 +164,57 @@ public Task<IActivityLogClient> CreateActivityLogClientAsync(CancellationToken c
...
+                workspaces = workspacesList;
+                if (workspaces.Count == 0)
+                {
+                    throw new InvalidOperationException("No log analytics workspaces were found");
+                }
+            }
+
+            // Verify there are not too many resources
+            if (resources.Count > MaxNumberOfResourcesInQuery)
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: >resources [](start = 16, length = 9)

workspaces

File: src/SDK/SmartDetectorsShared/Clients/TooManyResourcesInQueryException.cs [hunk 8]
```diff
@@ -8,49 +8,48 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.Clients
...
     using System.Runtime.Serialization;
-    using Microsoft.Azure.Monitoring.SmartDetectors.Presentation;
 
     /// <summary>
-    /// This exception is used to handle cases where query client information could not be created by the <see cref="IQueryRunInfoProvider"/>.
+    /// This exception is used to handle cases where there are too many resources in the query.
     /// </summary>
     [Serializable]
-    public class QueryClientInfoProviderException : Exception
+    public class TooManyResourcesInQueryException : Exception
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/TooManyResourcesInQueryException.cs]: >Exception [](start = 52, length = 9)

create a base class for AnalysisServicesFactoryException, and have both this and the other exception inherit from it

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 9]
```diff
@@ -143,23 +164,57 @@ public Task<IActivityLogClient> CreateActivityLogClientAsync(CancellationToken c
...
+                        this.subscriptionIdToWorkspaces[subscriptionId] = subscriptionWorkspaces;
+                    }
+
+                    workspacesList.AddRange(subscriptionWorkspaces);
+                }
+
+                workspaces = workspacesList;
+                if (workspaces.Count == 0)
+                {
+                    throw new InvalidOperationException("No log analytics workspaces were found");
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: >InvalidOperationException [](start = 30, length = 25)

change to use your own exception type

**Comment by EranGon** [comment]
Increase nuget versions!

File: src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs [hunk 10]
```diff
@@ -143,23 +164,57 @@ public Task<IActivityLogClient> CreateActivityLogClientAsync(CancellationToken c
...
+                if (workspaces.Count == 0)
+                {
+                    throw new TelemetryDataClientCreationException("No log analytics workspaces were found");
+                }
             }
 
-            // Verify that the resource IDs are not empty
-            if (!runInfo.ResourceIds.Any())
+            // Verify there aren't too many resources
+            if (workspaces.Count > MaxNumberOfResourcesInQuery)
```
[the remark this answers is not in the document]
Comment (melutzky) [reply src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs]: Yes

---
In reply to: [245566506](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/35#discussion_r245566506) [](ancestors = 245566506)

**Pull request merged**